### PR TITLE
Integration specs for views and fix n+1 problems

### PR DESCRIPTION
**Pull request by eudondian**
**### Integration Specs for Views and Fix n+1 Problems**

https://github.com/eudondian/Blog-Haven/assets/90483434/a5d31c67-1fb6-43b2-a67a-d98d921b74e6 (Before n+1)
https://github.com/eudondian/Blog-Haven/assets/90483434/a642ba97-9d04-4d62-bc3e-faa947365b0b (After n+1)

Used Capybara to write integration tests for each view in the project.

### CODE REVIEW

File: spec/integration/posts_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
+require 'rails_helper'
+
+RSpec.describe 'User posts', type: :system, js: true do
```
Comment (tugsem) [inline spec/integration/posts_index_spec.rb:3]: - [x] Please add a test that asserts you have a pagination button on this page as per the requirement. The button doesn't need to be functional at this point. The key point is that button is only visible ' if there are more posts than fit on the view.'

**Review by tugsem — CHANGES_REQUESTED** [review]
##  STATUS: REQUIRED CHANGES ♻️
Hello @eudondian  and @Timbar09 👋 , 

Good job so far! 👏 
There are some issues that you still need to work on to go to the next project but you are almost there!

Alt Text
## Highlights 🚀 
✔️all tests except one are implemented successfully
✔️capybara gem is used
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/integration/posts_index_spec.rb (hunk 1 above)
Comment (Timbar09) [reply spec/integration/posts_index_spec.rb:3]: Thank you for the feedback, @tugsem. We'll implement the changes shortly.

**Review by mckent05 — APPROVED** [review]
Hi @eudondian, 

# STATUS: APPROVED ✔️ ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations and well done! 🎉 
approved


## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👏 👏 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**